App.tsx

```diff
@@ -1,87 +1,101 @@
-import { StatusBar } from 'expo-status-bar';
 import { SafeAreaProvider } from 'react-native-safe-area-context';
-
 import useCachedResources from './hooks/useCachedResources';
 import useColorScheme from './hooks/useColorScheme';
-import Navigation from './navigation';
 import { ImageBackground } from 'react-native';
 import WelcomeScreen from './screens/WelcomeScreen';
-
 import { useFonts } from 'expo-font';
 import LoginStudentScreen from './screens/LoginStudentScreen';
-import { NavigationContainer, useNavigation } from '@react-navigation/native';
+import { NavigationContainer } from '@react-navigation/native';
 import { createNativeStackNavigator } from '@react-navigation/native-stack';
-
-import ActiveInScreen from './screens/ActiveInScreen';
-import ActiveInScreen2 from './screens/ActiveInScreen_2';
 import { RootStackParamList } from './types';
 import StudentInform from './screens/StudentInform';
 import StudentInform2 from './screens/StudentInform2';
-const image = require('./assets/images/background.png');
+import HomeScreen from './screens/HomeScreen';
+
+import * as SecureStore from 'expo-secure-store';
+import backendFetch from './lib/backendFetch';
+import { useEffect, useState } from 'react';
+
+const getToken = () => {
+  return SecureStore.getItemAsync('login-token');
+};
 
-let isLoggedIn = false; // For testing, will require further programming
+// Still TBD on how we do this
+// and if its needed.
+const validateToken = () => {
+  getToken().then(async token => {
+    if (token == (await backendFetch('GET', 'api-auth-token'))) {
+    }
+  });
+};
+
+let isLoggedIn = false;
+getToken().then(token => {
+  console.log('hi2');
+  if (token) {
+    isLoggedIn = true;
+  }
+});
```

Comment:
So, isLoggedIn will always be false during render? I'm a bit confused as to what this is supposed to do

Also left a console.log in there

Reply:
If token exists
isloggedin value gets overridden.

ezpz

Reply:
Correct, however, this will not cause a re-render, since it's not a variable initialized with useState()

Reply:
fixed in [7d357ce](https://github.com/TeamBAMI/frontend/pull/12/commits/7d357ce58dbb295e96e7563a0e0f06b6a7d72658)